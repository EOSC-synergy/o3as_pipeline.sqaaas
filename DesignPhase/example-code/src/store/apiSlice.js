// Import the RTK Query methods from the React-specific entry point
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'

// Define our single API slice object
export const apiSlice = createApi({
  // The cache reducer expects to be added at `state.api` (already default - this is optional)
  reducerPath: 'api',
  
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.o3as.fedcloud.eu/api/v1' }),
  
  // The "endpoints" represent operations and requests for this server
  endpoints: builder => ({
    
    getModels: builder.query({
      query: () => '/models'
    }),
    getPlots: builder.query({
      query: () => '/plots'
    }),
    getRawPlotData: builder.mutation({
      query: payload => ({
        // /data/tco3_return?begin=1959&end=2100&lat_min=-90&lat_max=90
        url: `/data/${payload.plotType}?begin=1959&end=2100&lat_min=${payload.lat_min}&lat_max=${payload.lat_max}`,
        method: "POST",
        body: payload.modelList,
      })
    }),
    
    getBuiltPlotData: builder.mutation({
      query: payload => ({
        
        url: `/plots/${payload.plotType}?begin=1959&end=2100&lat_min=${payload.lat_min}&lat_max=${payload.lat_max}`,
        method: "POST",
        body: payload.modelList,
      })
    }),
    
  })
})

// Export the auto-generated hook for the query endpoints
export const { 
  useGetPostsQuery,
  useGetPostQuery,
  useGetBuiltPlotDataMutation,
  useGetRawPlotDataMutation,
 } = apiSlice