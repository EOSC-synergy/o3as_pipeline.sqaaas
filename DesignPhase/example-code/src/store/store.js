import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice'
import modelsReducer from './modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})