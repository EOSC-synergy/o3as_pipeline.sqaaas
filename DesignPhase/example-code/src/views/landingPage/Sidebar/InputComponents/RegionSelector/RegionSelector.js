import React, { useState } from "react";
import LatitudeBandSelector from "../LatitudeBandSelector/LatitudeBandSelector";
import LocationSelector from "../LatitudeBandSelector/LatitudeBandSelector";

/**
 * enables the user to select / deselect regions as well as entering a private region {@link LatitudeBandSelector}

 * @param {Object} props
 * @param {Array} props.selectedRegions - the currently selected regions
 * @param {Function} props.error - used to report error functions
 * @returns {JSX} 
 */
function RegionSelector(props) {

    let i = props.selectedRegions;
    i = props.error;

    i = props.customRegion;
    i = props.changeCustomRegion;

    // TODO -> API Call: ist das in der API verfÃ¼gbar? sonst in einer config json "hardcoden"
    const getDefaultRegions = () => {

    }

    return (
        <>
            <LatitudeBandSelector />
        </>
    );
}

export default RegionSelector;