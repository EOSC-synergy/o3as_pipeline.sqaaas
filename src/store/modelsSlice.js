import { createSlice } from "@reduxjs/toolkit";

const STATISTICAL_VALUES = ["mean", "median", "derivative", "percentile"];

const initialState = {
    // currently active plot
    plotId: "tco3_zm",
    settings: {
        "tco3_zm": { 
            // this objects holds key-value-pairs, the keys being the model-group 
            // identifier and the values being the settings for each group 
            "all": { 
                // model group storing all information until it is possible 
                // to implement more model groups
                name: "All OCTS models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        color: "#000000",
                        plotStyle: "solid",
                        isVisible: true, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        },
        "tco3_return": {
            "all": { 
                name: "All Return/Recovery models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        hidden: false, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                        color: "#000000",
                        plotStyle: "solid",
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        }
    }
}

const modelsSlice = createSlice({
    name: "models",
    initialState,
    reducers: {
        // set models
        // add models

        addModels(state, action) { 
            const { groupId, newModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            // Add model to modelList if it is not already included
            for (var i = 0; i < newModelList.length; i++) {
                if (activeModelList.indexOf(newModelList[i]) !== -1) {
                    continue;
                }

                activeModelList.push(newModelList[i]);
            }

            // ---------------------------------------------------------
            // TODO: Add Model information to the model lookup table
            // ---------------------------------------------------------


        }, // expects groupID, list of models 
        removeModels(state, action) { 
            const { groupId, removeModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            for (var i = 0; i < removeModelList.length; i++) {
                // Remove model from modelList
                activeModelList = activeModelList.filter(item => item !== removeModelList[i]);

                // Delete model object from models
                delete selectedGroup.models[removeModelList[i]];
            }

        }, // expects groupdID, list of models
        updatedModelGroup(state, action) { }, // implementing this requires more knowledge about the UI
        addedModelGroup(state, action) { },
        setVisibility(state, action) { 
            const { groupID, modelID, isVisible } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            } 
            activeModel.isVisible = isVisible;
        },
        setStatisticalValuesIncluded(state, action) { // this is for an individual model
            // svType has to be a string of the form: 
            // mean | derivative | median | percentile
            const { groupID, modelID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            }
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeModel[svType] = isIncluded;
        },
        setStatisticalValueForGroup(state, action) { // this is for an entire group
            const { groupID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeSettings[groupID].visibileSV[svType] = isIncluded;
        },
    }
})

export const { 
    addModels,
    removeModels,
    updatedModelGroup,
    addedModelGroup,
    setVisibility, 
    setStatisticalValuesIncluded, 
    setStatisticalValueForGroup
} = modelsSlice.actions;

export default modelsSlice.reducer;

export const selectCurrentModelGroups = state => state.models[state.plot.plotId];
export const selectCurrentModelGroup = (state, groupId) => state.models[state.plot.plotId][groupId];