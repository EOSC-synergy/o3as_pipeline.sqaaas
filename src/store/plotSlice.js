import { createSlice } from "@reduxjs/toolkit"

/**
 * The initial state of the plotSlice defines the data structure in the 
 * store. Each plot has its own settings i.e. title, location etc.
 */
const initialState = {
    
    plotId: "tco3_zm", // currently active plot
    // maps plotids to their settings
    settings: {
        "tco3_zm": {
            name: "OCTS", // should show up in the drop down menu
            title: "OCTS Plot", // the title shown in the apexcharts generated chart
            location: {
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 1960,
                maxX: 2100,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        },
        "tco3_return": {
            name: "Return/Recovery",
            title: "Return/Recovery Plot",
            location: { // custom user defined region
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 0,
                maxX: 0,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        }
    }
}

/**
 * The plotSlice is generated by the redux toolkit. The reducers are defined here
 * and the corresponding actions are auto-generated.
 */
const plotSlice = createSlice({
    name: "plot",
    initialState,
    reducers: {
        setActivePlotId(state, action) { // e.g. dispatch(setActivePlotId({id: "tco3_zm"}))
            const { id } = action.payload
            state.plotId = id
        }, 
        setTitle(state, action) { // e.g. dispatch(setActivePlotId({title: "OCTS Plot"}))
            const { title } = action.payload
            state.settings[state.plotId].title = title
        },
        setLocation(state, action) { 
            const {minLat, maxLat} = action.payload
            const location = state.settings[state.plotId].location
            location.minLat = minLat
            location.maxLat = maxLat
        },
        setDisplayXRange(state, action) { 
            const {minX, maxX} = action.payload
            const displayXRange = state.settings[state.plotId].displayXRange
            displayXRange.minX = minX
            displayXRange.maxX = maxX
        },
        setDisplayYRange(state, action) { 
            const {minY, maxY} = action.payload
            const displayYRange = state.settings[state.plotId].displayYRange
            displayYRange.minY = minY
            displayYRange.maxY = maxY
        },
        setMonths(state, action) { 
            state.settings[state.plotId].months = action.payload.months
        },
    }   
})


/**
 * The here listed actions are exported and serve as an interface for
 * the view (our react components).
 */
export const {
    setActivePlotId,
    setTitle,
    setLocation,
    setDisplayXRange,
    setDisplayYRange,
    setMonths,
} = plotSlice.actions

/**
 * The reducer combining all reducers defined in the plot slice. 
 * This has to be included in the redux store, otherwise dispatching 
 * the above defined actions wouldn't trigger state updates.
 */
export default plotSlice.reducer

/**
 * This selectors allows components to select the current plot id
 * from the store. The plot id is a string using the same naming as the
 * o3as api e.g. tco3_zm or tco3_return
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot id
 */
export const selectPlotId = state => state.plot.plotId

/**
 * This selectors allows components to select the current plot name
 * from the store. The plot name is the humand readable version specifying 
 * the plot e.g. OCTS or Return/Recovery
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot name
 */
export const selectPlotName = state => state.plot.settings[state.plot.plotId].name

/**
 * This selectors allows components to select the current plot title
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot title
 */
export const selectPlotTitle = state => state.plot.settings[state.plot.plotId].title

/**
 * This selectors allows components to select the current plot location
 * from the store. The location is an object containg a minLat and maxLat attribute.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current location that includes a minLat and maxLat attribute.  
 */
export const selectPlotLocation = state => state.plot.settings[state.plot.plotId].location

/**
 * This selectors allows components to select the current x range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minX and maxX
 */
export const selectPlotXRange = state => state.plot.settings[state.plot.plotId].displayXRange

/**
 * This selectors allows components to select the current y range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minY and maxY
 */

export const selectPlotYRange = state => state.plot.settings[state.plot.plotId].displayYRange

/**
 * This selectors allows components to select the current selected months
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {array} array of strings describing the current selected months
 */

export const selectPlotMonths = state => state.plot.settings[state.plot.plotId].months
