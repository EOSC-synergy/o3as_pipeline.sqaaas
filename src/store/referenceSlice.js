import { createSlice } from "@reduxjs/toolkit"

const initialState = {
    "OCTS": {
        year: 1980,
        model: "CCMI-1_ACCESS_ACCESS-CCM-refC2",
        visibile: false,
        offsetApplied: false,
    }
}

const referenceSlice = createSlice({
    name: "reference",
    initialState,
    reducers: {
        setReferenceYear(state, action) {
            const {year} = action.payload;
            state.OCTS.year = year;
         },
        setReferenceModel(state, action) { 
            const {model} = action.payload;
            state.OCTS.model = model;
        },
        setVisibile(state, action) { },
        setOffsetApplied(state, action) { },
    }
})

export const {
    setReferenceYear,
    setReferenceModel,
    setVisibile,
    setOffsetApplied,
} = referenceSlice.actions

export default referenceSlice.reducer

export const selectCurrentReferenceSettings = state=> state.reference[state.plot.plotId]
