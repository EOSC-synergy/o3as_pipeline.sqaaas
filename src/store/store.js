import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice/plotSlice'
import modelsReducer from './modelsSlice/modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})