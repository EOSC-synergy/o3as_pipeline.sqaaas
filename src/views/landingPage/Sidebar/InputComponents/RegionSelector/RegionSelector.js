import React from "react";
import LatitudeBandSelector from "../LatitudeBandSelector/LatitudeBandSelector";
import {Box, Checkbox, FormControlLabel} from "@mui/material";
import {useDispatch, useSelector} from "react-redux";
import {selectPlotRegions, setRegions} from "../../../../../store/plotSlice/plotSlice";

/**
 * enables the user to select / deselect regions as well as entering a private region {@link LatitudeBandSelector}
 * @todo add redux connection
 * @param {Object} props
 * @param {function} props.reportError - used to report error functions
 * @returns {JSX.Element}
 */
function RegionSelector(props) {
    /**
     * A dispatch function to dispatch actions to the redux store.
     */
    const dispatch = useDispatch()

    /**
     * An array containing the selected regions.
     *
     * Examples:
     * If the first region is selected the array would have the following form: [0]
     * If the second and fifth region are selected the array would have the following form: [1, 4]
     */
    const selectedRegions = useSelector(selectPlotRegions);

    /**
     * Handles the change if a region is clicked (selected/deselected).
     *
     * @param {number} regionIdx The index of the region that was clicked.
     */
    const handleRegionChecked = (regionIdx) => {
        let regionCpy = [...selectedRegions];
        if (regionCpy.includes(regionIdx)) {
            regionCpy = regionCpy.filter((m) => m !== regionIdx);
        } else {
            regionCpy.push(regionIdx);
        }
        // Dispatch region checked
        dispatch(setRegions({ regions: regionCpy.sort((a, b) => a - b)}));
    }

    /**
     * Gets default regions that are available in the return recovery plot.
     * @todo connect to api
     */
    const getDefaultRegions = () => {
        return ["test1", "test21231", "test3123123123"];
    }

    return (
        <>
            <LatitudeBandSelector />

            <Box sx={{paddingLeft: '8%', paddingRight: '8%', alignItems: "left", display: "flex", flexDirection: "column"}}>
                {
                    getDefaultRegions().map((r, idx) => (
                        <React.Fragment key={idx}>
                            <FormControlLabel
                                label={r}
                                control={
                                    <Checkbox
                                        checked={selectedRegions.includes(idx)}
                                        onClick={() => handleRegionChecked(idx) }
                                    />
                                }
                            />
                        </React.Fragment>
                    ))
                }
            </Box>
        </>
    );
}

export default RegionSelector;