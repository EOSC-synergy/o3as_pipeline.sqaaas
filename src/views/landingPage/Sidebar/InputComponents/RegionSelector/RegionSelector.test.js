import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux'
import RegionSelector from './RegionSelector';
import { createTestStore } from "../../../../../store/store"

let store
beforeEach(() => {
    store = createTestStore();
});

it('renders without crashing', () => {
    render(<>
        <Provider store={store}>
            <RegionSelector reportError={() => {}} />
        </Provider>
    </>)
});

it('renders correctly', () => {

    const { container } = render(<>
        <Provider store={store}>
            <RegionSelector reportError={() => {}} />
        </Provider>
    </>);

    expect(container).toMatchSnapshot();
});