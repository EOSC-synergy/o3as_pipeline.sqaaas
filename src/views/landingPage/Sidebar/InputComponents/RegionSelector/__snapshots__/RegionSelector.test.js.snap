// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1ws8119-MuiDivider-root"
    role="separator"
  >
    <span
      class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        LATITUDE BAND
      </p>
    </span>
  </div>
  <div
    class="MuiBox-root css-1n3sbe4"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary css-vwjx0e-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="latitudeBandSelectorLabel latitudeBandSelector"
        class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="latitudeBandSelector"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="-90,90"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-173wfxe"
        >
          <span>
            LatitudeBand
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiBox-root css-11wo1kh"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        test1
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        test21231
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        test3123123123
      </span>
    </label>
  </div>
</div>
`;
